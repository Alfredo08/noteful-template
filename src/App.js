import React from 'react';
import {Route, Link} from 'react-router-dom';
import Profile from './Profile';
import AppContext from './AppContext';
import HobbyForm from './HobbyForm';

class App extends React.Component{

  state = {
    hobbies : ['program', 'dance', 'sing'],
    name : "Danielle"
  }

  addHobby = ( newHobby ) => {
    this.setState({
      hobbies : [...this.state.hobbies, newHobby]
    })
  }

  render(){
    const contextValues = {
      hobbies : this.state.hobbies,
      name : this.state.name,
      addHobby : this.addHobby
    };

    return(
      <AppContext.Provider value={contextValues}>
        <h1>
          Hello there welcome to our website
        </h1>
        <nav>
          <Link to="/profile">
            Profile
          </Link>
          <Link to="/add-hobby">
            Add Hobby
          </Link>
        </nav>
        
        <Route path="/profile" component={ Profile }/>
        <Route path="/add-hobby" component={ HobbyForm }/>
      </AppContext.Provider>
    );
  }
}

export default App;
