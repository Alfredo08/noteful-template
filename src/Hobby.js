import React from 'react';

function Hobby( props ){
    return(
        <div>
            <h2>
                I really like to {props.hobby}
            </h2>
        </div>
    );
}

export default Hobby;