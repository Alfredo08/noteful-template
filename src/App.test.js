import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Hobby from './Hobby';

// Smoke tests
it( "Should load the App component correctly", () => {
  let testContainer = document.createElement( 'div' );
  ReactDOM.render( <App />, testContainer );
  ReactDOM.unmountComponentAtNode( testContainer );
});

it( "Should load the Hobby component correctly", () => {
  let testContainer = document.createElement( 'div' );
  ReactDOM.render( <Hobby hobby="Sing"/>, testContainer );
  ReactDOM.unmountComponentAtNode( testContainer );
})

// Snapshot tests
it( 'Renders the virtual Hobby in the UI correctly', () => {
  let testContainer = renderer.create( <Hobby hobby="Sing" /> ).toJSON();
  expect( testContainer ).toMatchSnapshot();
});

/*

it( 'Renders the virtual Note in the UI correctly', () => {
  let testContainer = renderer.create( <Note  note={dummyNote}/> ).toJSON();
  expect( testContainer ).toMatchSnapshot();
});

*/