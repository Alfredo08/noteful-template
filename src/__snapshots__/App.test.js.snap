// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the virtual Hobby in the UI correctly 1`] = `
<div>
  <h2>
    I really like to 
    Sing
  </h2>
  <p />
</div>
`;
